homeassistant/components/image_processing/facebox_face_detect.py

```diff
@@ -0,0 +1,90 @@
+"""
+Component that will perform facial detection via a local facebox instance.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/image_processing.facebox_face_detect
+"""
+import base64
+import requests
+import logging
+import voluptuous as vol
+
+from homeassistant.core import split_entity_id
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.image_processing import (
+    PLATFORM_SCHEMA, ImageProcessingFaceEntity, CONF_SOURCE, CONF_ENTITY_ID,
+    CONF_NAME)
+from homeassistant.const import (CONF_IP_ADDRESS, CONF_PORT)
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_IP_ADDRESS): cv.string,
+    vol.Required(CONF_PORT): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the classifier."""
+    entities = []
+    for camera in config[CONF_SOURCE]:
+        entities.append(FaceboxFaceDetectEntity(
+            config[CONF_IP_ADDRESS],
+            config[CONF_PORT],
+            camera[CONF_ENTITY_ID],
+            camera.get(CONF_NAME)
+        ))
+    add_devices(entities)
+
+
+class FaceboxFaceDetectEntity(ImageProcessingFaceEntity):
+    """Perform a classification via a Facebox."""
+
+    def __init__(self, ip, port, camera_entity, name=None):
+        """Init with the API key and model id"""
+        super().__init__()
+        self._url = "http://{}:{}/facebox/check".format(ip, port)
+        self._camera = camera_entity
+        if name:
+            self._name = name
+        else:
+            self._name = "Facebox {0}".format(
+                split_entity_id(camera_entity)[1])
+        self.total_faces = 0
+        self.faces = []
+
+    def process_image(self, image):
+        """Process an image."""
+        response = {}
+        try:
+            response = requests.post(
+                self._url,
+                json=self.encode_image(image),
+                timeout=30
+                ).json()
+        except requests.exceptions.ConnectionError:
+            _LOGGER.error("ConnectionError: Is Facebox running?")
+            response['success'] = False
+
+        if response['success']:
+            self.total_faces = response['facesCount']
+            self.faces = response['faces']
+
+        else:
+            self.total_faces = "Request_failed"
```

Comment:
This would change the type of total faces from integer to string. That will make it hard to use templates for this attribute.

I suggest that we don't update the total faces if there's an error. Or set it to `0`.

Reply:
Or None? I don’t want to have users think a classification has been successful and that the face count is zero, when in face the classification was unsuccessful

Reply:
`None` is good.